modules/navier_stokes/src/kernels/INSADMomentumViscous.C

```diff
@@ -57,15 +57,16 @@ INSADMomentumViscous::qpViscousTerm()
 ADRealVectorValue
 INSADMomentumViscous::qpAdditionalRZTerm()
 {
+  ADRealVectorValue ret;
+  auto & extra_term = ret(_rz_radial_coord);
+
   // Add the u_r / r^2 term. There is an extra factor of 2 for the traction form
-  ADReal resid = _mu[_qp] * _u[_qp](_rz_radial_coord);
+  const auto & r = _ad_q_point[_qp](_rz_radial_coord);
+  extra_term = _mu[_qp] * _u[_qp](_rz_radial_coord) / (r * r);
   if (_form == "traction")
-    resid *= 2.;
+    extra_term *= 2.;
 
-  if (_use_displaced_mesh)
```

Comment:
this was just an optimization right

Reply:
Some unknown degree of optimization yes. My hope is that multiplying dual numbers without derivatives is not too expensive in the scheme of things